xmodule/capa_block.py

```diff
@@ -1433,6 +1434,19 @@ def closed(self):
         """
         Is the student still allowed to submit answers?
         """
+        # Checking if the course is archived
+        from xmodule.modulestore.django import modulestore
+        try:
+            if isinstance(self.course_id, CourseKey):
+                course = modulestore().get_course(self.course_id)
+                closed_date = (
+                    course.end + self.graceperiod if self.graceperiod is not None else course.end
+                )
+                if closed_date and datetime.datetime.now(utc) > closed_date:
+                    return True
+        except AttributeError:
+            pass
```

Comment:
I think we might have other solutions to this problem. e.g. How edX is checking for archived courses on other places. I also think this is something that would need a product review. 

However, What I would try to do here is:

**Option 1:**

- Check how archived courses are being detected in other places, One example is [HomePage Course View](https://github.com/openedx/edx-platform/blob/master/cms/djangoapps/contentstore/rest_api/v1/views/home.py#L157) that uses some functionality to check for archived courses. The API eventually ends up checking [course.has_ended()](https://github.com/openedx/edx-platform/blob/master/cms/djangoapps/contentstore/views/course.py#L819) to check for an archived course.
    - If we go for this solution, We should also use that same has_ended method instead of writing our own code
    - If we need to add the grace period as well, We should ideally move that to `has_ended` method. However, All the places I see that check for grace period are related to grading only.


**Option 2:**

 If we can't go with option#1, We should move this code into another function and call it here just like how we are using `self.is_past_due():` below. This new function can check both `has_ended` from the course and graceperiod if needed.